bool[] appear
int16[] point
float64[] cx
float64[] cy
float64[] area
